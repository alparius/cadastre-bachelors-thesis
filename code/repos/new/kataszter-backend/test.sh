#!/bin/bash

rm -f coverage.txt
echo 'mode: atomic' > coverage.txt

for d in $(go list ./... | grep -v testing); do
    go test -v -p 1 -race -covermode=atomic -coverprofile=coverage.out -coverpkg=$(go list ./...| grep -v testing | tr "\n" ",") $d
    if [ -f coverage.out ]; then
        tail -n +2 coverage.out >> coverage.txt || exit 255
        rm coverage.out
    fi
done
