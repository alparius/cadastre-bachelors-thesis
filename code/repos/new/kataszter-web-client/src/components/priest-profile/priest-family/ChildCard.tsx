import React from "react";
import { Button, Card, Header, List } from "semantic-ui-react";

import IChild from "../../../data/priest/Child";
import CardRow from "../CardRow";
import CardRowForDate from "../CardRowForDate";

interface IChildCardProps {
    child: IChild;
    index: number;
    editMode: boolean;
    handleDelete: (index: number) => void;
    handleChangeOnChild: (labelName: string, value: any, index: number) => void;
}

const ChildCard: React.FC<IChildCardProps> = (props: IChildCardProps) => {
    const { child, index, editMode, handleDelete, handleChangeOnChild } = props;

    const handleOnClick = () => {
        handleDelete(index);
    };

    return (
        <Card fluid>
            <Card.Content>
                <Card.Content>
                    {editMode ? (
                        <Button id="deleteButton" floated="right" icon="trash" color="red" inverted onClick={handleOnClick} />
                    ) : (
                        <Card.Header id="header">
                            <Header as="h2">{child.name}</Header>
                        </Card.Header>
                    )}
                </Card.Content>
                <Card.Description>
                    <List>
                        {editMode && (
                            <List.Item>
                                <CardRow labelName="name" index={index} data={child.name} editMode={editMode} setValue={handleChangeOnChild} />
                            </List.Item>
                        )}
                        <List.Item>
                            <CardRow
                                labelName="birthPlace"
                                index={index}
                                data={child.birthPlace}
                                editMode={editMode}
                                setValue={handleChangeOnChild}
                            />
                        </List.Item>
                        <List.Item>
                            <CardRowForDate
                                labelName="birthDate"
                                index={index}
                                data={child.birthDate}
                                editMode={editMode}
                                setValue={handleChangeOnChild}
                            />
                        </List.Item>
                    </List>
                </Card.Description>
            </Card.Content>
        </Card>
    );
};

export default ChildCard;
