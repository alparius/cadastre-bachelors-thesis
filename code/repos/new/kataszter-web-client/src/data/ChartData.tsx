interface IChartData {
    key: any;
    data: any;
    counter: number;
}

export default IChartData;
