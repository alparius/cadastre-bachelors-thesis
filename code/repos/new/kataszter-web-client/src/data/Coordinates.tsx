interface ICoordinates {
    lat: number;
    lng: number;
}

export let NewCoordinates: ICoordinates = {
    lat: 0,
    lng: 0
};

export default ICoordinates;
