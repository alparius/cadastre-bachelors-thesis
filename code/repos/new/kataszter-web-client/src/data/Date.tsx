interface IDate {
    year: number;
    month: number;
    day: number;
    comment: string;
}

export let NewDate: IDate = {
    year: 0,
    month: 0,
    day: 0,
    comment: ""
};

export default IDate;
