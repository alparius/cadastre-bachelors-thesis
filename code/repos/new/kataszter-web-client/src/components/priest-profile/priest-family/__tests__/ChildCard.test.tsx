/*===================
  ChildCard.unit.test.tsx
 ===================*/
import React from "react";

import IChild from "../../../../data/priest/Child";
import { NewTestChild } from "../../../../util/test/data/test-data";
import { createComponentWithIntlAndRouter, mountWithIntl } from "../../../../util/test/test-wrapper";
import ChildCard from "../ChildCard";

const handlers = {
    handleDelete: jest.fn(),
    handleChangeOnChild: jest.fn()
};

describe("ChildCard element", () => {
    afterEach(() => {
        handlers.handleDelete.mockClear();
        handlers.handleChangeOnChild.mockClear();
    });

    it("should render a card with the child data", () => {
        const child: IChild = NewTestChild;
        const container = createComponentWithIntlAndRouter(<ChildCard child={child} index={0} editMode={false} {...handlers} />).toJSON();
        expect(container).toMatchSnapshot();
    });

    it("should call handleDelete if deleteButton pressed", () => {
        const child: IChild = NewTestChild;
        const container = mountWithIntl(<ChildCard child={child} index={0} editMode {...handlers} />);
        container
            .find("#deleteButton")
            .at(0)
            .simulate("click");

        expect(handlers.handleDelete).toHaveBeenCalledTimes(1);
    });

    it.each([
        [0, "1", "2"],
        [1, "1", "2"],
        [2, "1", "2"]
    ])("should  call handleChangeOnChild on %d. input. After first change: %s, After second change: %s", (inputNr, firstChange, secondChange) => {
        const child: IChild = NewTestChild;
        const container = mountWithIntl(<ChildCard child={child} index={0} editMode {...handlers} />);
        const input = container.find("input").at(Number(inputNr));
        input.simulate("change", { target: { value: firstChange } });
        input.simulate("change", { target: { value: secondChange } });
        expect(handlers.handleChangeOnChild).toHaveBeenCalledTimes(2);
    });
});
