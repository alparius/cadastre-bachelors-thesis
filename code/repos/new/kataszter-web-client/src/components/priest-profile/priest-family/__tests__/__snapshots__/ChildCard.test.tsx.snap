// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ChildCard element should render a card with the child data 1`] = `
<div
  className="ui fluid card"
  onClick={[Function]}
>
  <div
    className="content"
  >
    <div
      className="content"
    >
      <div
        className="header"
        id="header"
      >
        <h2
          className="ui header"
        >
          Gyerek
        </h2>
      </div>
    </div>
    <div
      className="description"
    >
      <div
        className="ui list"
        role="list"
      >
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Place of birth
                :
              </b>
               
              Krahorandia
            </div>
          </form>
        </div>
        <div
          className="item"
          onClick={[Function]}
          role="listitem"
        >
          <form
            className="ui form"
            onSubmit={[Function]}
          >
            <div
              className="inline field"
            >
              <b>
                Date of birth
                :
              </b>
               
              <span
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onMouseEnter={[Function]}
                onMouseLeave={[Function]}
              >
                2000.06.06
              </span>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</div>
`;
